### PR TITLE
Gulp publish flow

### PR DESCRIPTION
**Pull request by levithomason**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -25,19 +45,21 @@ var distMinName = changeCase.pascalCase(pkg.name) + '.min.js';
...
-      'webpack',
-      'doc'
-    ],
+    'webpack',
     cb
   )
 });
 
 gulp.task('clean-dist', function(cb) {
+  var del = require('del');
```
Comment (levithomason) [inline gulpfile.js]: Moved non-gulp `require()`s into the tasks that need them.  This speeds up gulp's initialization.

**Comment by eanplatter** [comment]
༼ つ ◕_◕ ༽つ

**Comment by levithomason** [comment]
Why on :earth_americas: is this PR showing so many doc changes?!  Good night.

**Comment by TomFrost** [comment]
Whoa, looks like quite a few merge conflicts got committed verbatim in the docs.

File: gulpfile.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
-var argv = require('yargs').argv;
```
Comment (levithomason) [inline gulpfile.js]: Review this file

File: gulpfile.js [hunk 3]
```diff
@@ -68,13 +118,33 @@ gulp.task('webpack', function() {
...
+//
+
+gulp.task('clean-dist', function(cb) {
+  del(paths.dist, cb);
+});
+
+gulp.task('clean-docs', function(cb) {
+  del(paths.docs, cb);
+});
+
```
Comment (levithomason) [inline gulpfile.js]: Cleaning the docs is what caused the +120k removed lines :)  Old doc files were piling up!

**Comment by eanplatter** [comment]
☆*:.｡. o(≧▽≦)o .｡.:*☆

**Pull request merged**